src/averageAges.js

```diff
@@ -55,7 +72,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = (onlyWithSon)
+    ? people
+      .filter(person => person.sex === 'm'
+        && people.some(mom => mom.name === person.mother))
```

Comment:
move to the separate function
```suggestion
        && hasMother(people, person.mother))
```